File: frontend/src/domain/entities/Inspection.ts
import type { InspectionStatus } from '@/domain/types/inspection'

export class Inspection {
  readonly id: string
  readonly title: string
  status: InspectionStatus
  readonly createdAt: string // ISO 8601 UTC形式（ローカルで発行）
  updatedAt: string // ISO 8601 UTC形式（ローカルで発行）
  readonly description?: string

  constructor(
    id: string,
    title: string,
    status: InspectionStatus,
    createdAt: string,
    updatedAt: string,
    description?: string
  ) {
    this.id = id
    this.title = title
    this.status = status
    this.createdAt = createdAt
    this.updatedAt = updatedAt
    this.description = description
  }
}

File: backend/src/main/kotlin/com/localbridge/backend/entity/Inspection.kt
package com.localbridge.backend.entity

import jakarta.persistence.*
import java.time.Instant

@Entity
@Table(name = "inspections")
data class Inspection(
    @Id
    @Column(length = 36)
    val id: String,

    @Column(nullable = false, length = 255)
    val title: String,

    @Column(nullable = false, length = 50)
    @Enumerated(EnumType.STRING)
    val status: InspectionStatus,

    @Column(columnDefinition = "TEXT")
    val description: String? = null,

    @Column(name = "created_at", nullable = false)
    val createdAt: Instant = Instant.now(),

    @Column(name = "updated_at", nullable = false)
    val updatedAt: Instant = Instant.now()
)

enum class InspectionStatus {
    TODO,
    IN_REVIEW,
    DONE,
    CORRECTION_NEEDED
}
